src/test/java/com/coherentsolutions/aqa/java/web/makarevich/TableSortAndSearchTest.java

```diff
@@ -0,0 +1,46 @@
+package com.coherentsolutions.aqa.java.web.makarevich;
+
+import lombok.extern.slf4j.Slf4j;
+import com.coherentsolutions.aqa.web.makarevich.model.Employee;
+import com.coherentsolutions.aqa.web.makarevich.services.Employees;
+import org.openqa.selenium.support.ui.Select;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import com.coherentsolutions.aqa.web.makarevich.testbase.TestBase;
+
+import java.util.List;
+
+import static com.coherentsolutions.aqa.web.makarevich.constants.ByConstants.SHOW_ENTRIES_DROPDOWN;
+import static com.coherentsolutions.aqa.web.makarevich.constants.UrlConstants.TABLE_TEST_URL;
+
+@Slf4j
+public class TableSortAndSearchTest extends TestBase {
+
+    public static final String optionToBeSelected = "10";
+    public static final int age = 40;
+    public static final double salary = 150000.0;
+
+    @Test
+    public void testTableSortAndSearch() {
+        log.info("Open the URL: {}", TABLE_TEST_URL);
+        driver().get(TABLE_TEST_URL);
+
+        Select showEntriesDropdown = new Select(driver().findElement(SHOW_ENTRIES_DROPDOWN));
+        String selectedOption = showEntriesDropdown.getFirstSelectedOption().getText();
+        log.info("Set up the Show {} entries filter", optionToBeSelected);
+        if (!selectedOption.equals(optionToBeSelected)) {
+            showEntriesDropdown.selectByVisibleText(optionToBeSelected);
+        }
+        selectedOption = showEntriesDropdown.getFirstSelectedOption().getText();
+        Assert.assertEquals(optionToBeSelected, selectedOption, String.format("%s is not selected in Show entries dropdown", optionToBeSelected));
+
+        log.info("Sort employees by following criteria: age > {} and salary <= {}", age, salary);
+        List<Employee> sortedEmployeeList = new Employees(driver()).sortEmployees(age, salary);
+        Assert.assertFalse(sortedEmployeeList.isEmpty(), "Sorted employee list is empty!");
+
+        log.info("Print sorted employees");
```

Comment:
https://github.com/EugenMakarevich/SeleniumWebDriver/pull/4/files#r1335475676

Reply:
Deleted that part since it's not obligatory per test case